src/tiledbsoma_ml/dataset.py

```diff
@@ -5,218 +5,619 @@
-      3. The entire I/O buffer is shuffled.
+    3. IO-batching: shuffle-chunks are re-grouped into "IO batches" of size |io_batch_size|. If ``shuffle=True``, each
+       IO-batch is shuffled, then the corresponding ``X`` and ``obs`` rows are fetched from the underlying
+       ``Experiment``.
 
-    Put another way, we read randomly selected groups of observations from across all query results, concatenate
-    those into an I/O buffer, and shuffle the buffer before returning mini-batches. The randomness of the shuffle
-    is therefore determined by the ``io_buffer_size`` (number of rows read), and the ``shuffle_chunk_size``
-    (number of rows in each draw). Decreasing ``shuffle_chunk_size`` will increase shuffling randomness, and decrease I/O
-    performance.
+    4. GPU-batching (|Iterable|\[|Batch|\]): IO-batch tuples are re-grouped into "GPU batches" of size |batch_size|.
 
-    This class will detect when run in a multiprocessing mode, including multi-worker :class:`torch.utils.data.DataLoader`
-    and multi-process training such as :class:`torch.nn.parallel.DistributedDataParallel`, and will automatically partition
-    data appropriately. In the case of distributed training, sample partitions across all processes must be equal. Any
-    data tail will be dropped.
+    Shuffling support (in steps 2. and 3.) is enabled with the ``shuffle`` parameter, and should be used in lieu of
+    |DataLoader|'s default shuffling functionality. Similarly, |batch_size| should be used instead of |DataLoader|'s
+    default batching. |experiment_dataloader| is the recommended way to wrap an |ExperimentDataset| in a
+    |DataLoader|, as it enforces these constraints while passing through other |DataLoader| args.
+
+    Describing the whole process another way, we read randomly selected groups of ``obs`` coordinates from across all
+    |ExperimentAxisQuery| results, concatenate those into an I/O buffer, shuffle the buffer element-wise, fetch the full
+    row data (``X`` and ``obs``) for each coordinate, and send that on to PyTorch / the GPU, in batches. The randomness
+    of the shuffle is determined by:
+
+      - |shuffle_chunk_size|: controls the granularity of the global shuffle. ``shuffle_chunk_size=1`` corresponds to
+        a full global shuffle, but decreases I/O performance. Larger values cause chunks of rows to be shuffled,
+        increasing I/O performance (by taking advantage of data locality in the underlying |Experiment|) but decreasing
+        overall randomness of the yielded data.
+      - |io_batch_size|: number of rows to fetch at once (comprised of concatenated shuffle-chunks, and shuffled
+        row-wise). Larger values increase shuffle-randomness (by shuffling more "shuffle chunks" together), I/O
+        performance, and memory usage.
 
     Lifecycle:
         experimental
     """
 
+    layer_name: str
```

Comment:
same question as for _csr.py: why add these names to the Class-level, rather than just to instance objects?  Apologies - asking out of ignorance